src/python/grpcio/grpc/_server.py

```diff
@@ -329,6 +329,7 @@ def _look_for_request(self):
             request = self._state.request
             self._state.request = None
             return request
+        raise Exception('gRPC Error: Invalid _state')
```

Comment:
Let's not add this code? What's the inspection that we would have to suppress here? I suspect we should just suppress it.

Reply:
The inspection is if you are returning a value from at least one code path within a function (i.e. there exists at least one `return <expression>` statement), all code paths should be statically verifiable to return a value.  I think this is a valuable verification actually, and we shouldn't turn it off, as otherwise you would be silently returning `None` in a "non-void" function.  I believe an explicit exception is better suited to ensure we never visit unanticipated branches.  However, I acknowledge that I'm not sure what's the best exception type to raise. (or perhaps `assert False` instead?)

Reply:
Turned into `raise AssertionError()` instead